### PR TITLE
WEB-1335 Make Merlin's footer more app-like

### PR DESCRIPTION
**Pull request by mitchquik**
(brief description / PR title)

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1335

## Changes
- Changed newsletter heading to h4
- changed button to inline tertiary style
- added additional fieldrow wrapper to align inline
- removed List component from links, changed to Link component
- changed link styling
- Moved social icons to navigation menu
- Moved copyright to navigation menu

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- scroll to footer
- open menu to check content in navigation

### CODE REVIEW

**Comment by mitchquik** [comment]
Fixed merge conflicts with develop branch and updated CI test files. @jeffkamo back to you

File: web/app/containers/footer/_footer.scss [hunk 1]
```diff
@@ -9,57 +9,31 @@
...
 // ---
 
-.t-footer__copyright {
-    margin: 0;
-
-    color: #fff;
-    font-size: 12px;
-    line-height: $font-size;
+.t-footer__newsletter {
+    background: #fff;
```
Comment (gsaray) [inline web/app/containers/footer/_footer.scss]: I think you can remove this class. background is white by default so this is redundant.

File: web/app/containers/footer/partials/footer-newsletter-subscription.jsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ class FooterNewsletterSubscription extends React.Component {
         return (
             <div className="t-footer__newsletter u-padding-md u-padding-top-lg u-padding-bottom-lg">
                 <div>
-                    <h2 className="u-h2 u-margin-bottom-md">
+                    <h2 className="u-h4 u-margin-bottom-md u-text-all-caps">
```
Comment (gsaray) [inline web/app/containers/footer/partials/footer-newsletter-subscription.jsx]: let's use `u-text-uppercase` here. Cole has removed `u-text-all-caps` in another PR so this is gonna break when both PRs are merged

File: web/app/containers/navigation/container.jsx [hunk 3]
```diff
@@ -70,6 +80,31 @@ const Navigation = (props) => {
                 </HeaderBar>
 
                 <NavMenu itemFactory={itemFactory} />
+
+                <div>
+                    <div className="t-navigation__social">
```
Comment (gsaray) [inline web/app/containers/navigation/container.jsx]: you are basically repeating the NavigationSocialIcons implementation here. let's import `NavigationSocialIcons` from partials and use it here.

**Pull request merged**